### ISSUE TITLE
Inconsistent use of `rope_traditional`

### ISSUE THREAD

**Issue by armbues** [ml-explore/mlx-examples#170]
I have been trying to get conversion of HF models to work and when I finally got key mapping right, I ran into issues where the first generated tokens looked fine but were increasingly incoherent ending up in random tokens.

I tried converting the following models:
- Llama-2-13b-hf
- Mistral-7B-Instruct-v0.2

After a lot of debugging with the config files I found that setting `rope_traditional` to `False` fixed the issues. However, the LLM examples handle this parameter inconsistently in `ModelArgs`:
- In `llama.py` it exists but defaults to `True`, so needs to be manually set in `config.json`
- In `mistral.py` it's missing and when initializing `nn.RoPE` it's set to `True`
- In `lora/models.py` it's also missing and when initializing `nn.RoPE` it's set to `True`

Making this configurable would be the first step to make additional models work, but I'm wondering if there is a way to derive this setting when converting a model `config.json`.

**Comment by armbues** [comment]
After actually reading up on what the `traditional` flag in the MLX RoPE implementation is used for, it seems like this would be a bug in the implementation in combination with different models instead of a configuration error.

**Comment by awni** [comment]
Hi @armbues could you elaborate on the bug with `traditional` I don't follow.

Also the Mistra 7B model should be working now, there was a config error in the `rope_theta` param. I just tried it and it gives solid output.

I will also double check the 13b model.

**Comment by armbues** [comment]
Hi @awni, thanks for looking into this!

Here's an example output for `Mistral-7B-Instruct-v0.2` using `rope_traditional == True` and the default prompt from the example script:

`In the beginning the Universe was created.
 According to various religious and philosophical beliefs, the Universe was created. In science, particularly in the Abrahamic religions, the Universe was created by a deity or deities, depending on one's beliefs. In physics, the Universe began as an extremely hot, dense and uniformly distributed across space and time, the Big Bang theory suggests that the Universe began around 13,13, around 13,3,13,3, particularly monotheistic traditions, the Universe was created by a divine being, force, or deity, or God, or a supreme being, or a singularity, around 13, the Big Bang Theory, approximately13, around 13, around 13, about 13, around 13, approximately 13, around 13, around 13, roughly around 13, around 13, around 13, around 13, around 13, approximately 13, around 13, around 13, around 13, around13,3, around 13,3, roughly13, around13, approximately13, around 13,3,`

You can see what I tried to describe as the output getting slowly incoherent.

Same prompt for `Llama-2-13b` also with `rope_traditional == True`:

`In the beginning the Universe was created.
During the beginning, before man and creation, there was a single point of unlimited potential. Nothing.
[/color=#193E]
[/b]`

What follows are seemingly random tokens, so it seems to go off the rails faster.

When switching `rope_traditional` to `False`, the models output looks great.

(I've confirmed after updating to MLX 0.0.6 that I'm still seeing this behavior.)

**Comment by awni** [comment]
@armbues what commit hash of mlx-examples are you using? Also what is the value of `rope_theta` in the `model/config.json`?

I am not getting the same behavior as you with the default prompt, but it's important that you are using the latest mlx-examples and the latest version of the mistral instruct v2 model (which set rope_theta correctly). For reference here is the output I get with `python mistral.py`:
```
In the beginning the Universe was created. This has made a lot of people very upset and been widely regarded as a bad move.

Douglas Adams

# The Big Bang

The Big Bang theory is the prevailing cosmological model explaining the origin of the Universe. It is based on the observations that the Universe is expanding and that the light from distant galaxies is redshifted, indicating that they are moving away from us. The Big Bang theory proposes that the Universe originated from an extremely hot and dense state
------
```

**Comment by armbues** [comment]
@awni I updated to the latest version of mlx-examples and now `Mistral-7B-Instruct-v0.2` works fine. The `rope_theta` of that model was always `1000000.0` as in the source model.

The problems with Llama models actually look like an unrelated bug. I'll have to double-check the configurations of the models as it seems I converted them with earlier versions of different `convert.py` scripts.

**Comment by awni** [comment]
Yes maybe try the new `convert.py` scripts + new mlx-examples. If you still have issues with Llama 2 let me know!

**Comment by awni** [comment]
I'm going to close this for now, but I can reopen if needed.

**Comment by armbues** [comment]
@awni following up since I finally figured out what went wrong and it might be applicable to the recent commit for HF model support.

Back when I submitted the issue, I was trying to convert the HF models for `Llama-2-13b-hf` and `Mistral-7B-Instruct-v0.2` with my own scripts. I ended up using the safetensors variant since I didn't have to deal with sharding (for the 13b model).

The problems that I was seeing with RoPE were apparently caused by missing weights for `rope.freqs` in safetensors for both the models. When switching `rope_traditional` to `False` they worked more or less.
